xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIMEGLES.cpp

```diff
@@ -0,0 +1,219 @@
+/*
+ *      Copyright (C) 2007-2015 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "RendererDRMPRIMEGLES.h"
+
+#include "cores/VideoPlayer/DVDCodecs/DVDCodecUtils.h"
+#include "cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.h"
+#include "cores/VideoPlayer/VideoRenderers/RenderFactory.h"
+#include "ServiceBroker.h"
+#include "utils/log.h"
+
+CWinSystemGbmGLESContext *CRendererDRMPRIMEGLES::m_pWinSystem = nullptr;
+
+CRendererDRMPRIMEGLES::~CRendererDRMPRIMEGLES()
+{
+  for (int i = 0; i < NUM_BUFFERS; ++i)
+    DeleteTexture(i);
+}
+
+CBaseRenderer* CRendererDRMPRIMEGLES::Create(CVideoBuffer* buffer)
+{
+  if (buffer && dynamic_cast<CVideoBufferDRMPRIME*>(buffer))
+    return new CRendererDRMPRIMEGLES();
+
+  return nullptr;
+}
+
+bool CRendererDRMPRIMEGLES::Register(CWinSystemGbmGLESContext *winSystem)
+{
+  VIDEOPLAYER::CRendererFactory::RegisterRenderer("drm_prime_gles", CRendererDRMPRIMEGLES::Create);
+  m_pWinSystem = winSystem;
+
+  return true;
+}
+
+bool CRendererDRMPRIMEGLES::Configure(const VideoPicture &picture, float fps, unsigned int orientation)
+{
+  for (auto &texture : m_DRMPRIMETextures)
+    texture.Init(m_pWinSystem->GetEGLDisplay());
+
+  return CLinuxRendererGLES::Configure(picture, fps, orientation);
+}
+
+void CRendererDRMPRIMEGLES::ReleaseBuffer(int index)
+{
+  m_DRMPRIMETextures[index].Unmap();
+  CLinuxRendererGLES::ReleaseBuffer(index);
+}
+
+void CRendererDRMPRIMEGLES::AddVideoPicture(const VideoPicture& picture, int index, double currentClock)
+{
+  YUVBUFFER &buf = m_buffers[index];
+  if (buf.videoBuffer)
+    buf.videoBuffer->Release();
```

Comment:
This should not be needed as `Flush` in parent `LinusRendererGELS` calls `ReleaseBuffer` to clear buffers, and looks like this might miss an `Unmap`, better call `ReleaseBuffer` if you cant to keep this